patches/tModLoader/Terraria/Main.cs.patch

```diff
@@ -1712,28 +1713,34 @@
  					if (item.melee) {
  						int num3 = player[myPlayer].meleeCrit - player[myPlayer].inventory[player[myPlayer].selectedItem].crit + item.crit;
  						toolTipLine[numLines] = num3 + Lang.tip[5].Value;
-@@ -15152,6 +_,14 @@
+@@ -15152,8 +_,16 @@
  						toolTipLine[numLines] = num5 + Lang.tip[5].Value;
  						numLines++;
  					}
 +					*/
-+					// TODO: make this not inherently exclude summons. minions should be allowed to crit in the right circumstances, too
-+					if (!item.summon) {
++					
++					if (item.DamageType.AllowStandardCrits() && item.DamageType.ShowStatTooltipLine(player[myPlayer], "CritChance")) {
```

Comment:
I don't recall seeing any example mod comment or documentation explaining that you need to have both aspects set:
AllowCrits is true, and is showing that it has Crit Chance. 
Ensure that this behaviour is covered someplace.